db/disk/course.go

```diff
@@ -107,8 +113,8 @@ func (this *backend) GetCourse(courseID string) (*model.Course, error) {
 }
 
 func (this *backend) GetCourses() (map[string]*model.Course, error) {
-    this.lock.RLock();
-    defer this.lock.RUnlock();
+    common.ReadLock("all-courses");
```

Comment:
If we were going to go this route, then we would need to make a constant and add a prefix/suffix for the disk db.
Remember that the whole project will share the name space, so we want to be sure and not conflict.

However, I think the way to go here is to actually lock each one as we go.
You would need to make sure that each one gets unlocked.
There are several ways to do that, I would probably use an anonymous function so that I could continue to use `defer`.